src/App.jsx

```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [goods, setGoods] = useState(initialGoods);
+  const [sortAlphabeticallyLight, setSortAlphabeticallyLight] = useState(true);
+  const [sortByLengthLight, setSortByLengthLight] = useState(true);
+  const [reverseLight, setReverseLight] = useState(true);
+  const [resetLight, setResetLight] = useState(true);
+  const [ascending, setAscending] = useState(true);
+
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort();
+
+    setGoods(sortedGoods);
+    setAscending(true);
+    setSortAlphabeticallyLight(false);
+    setSortByLengthLight(true);
+    setReverseLight(true);
+    setResetLight(true);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(sortedGoods);
+    setAscending(true);
+    setSortAlphabeticallyLight(true);
+    setSortByLengthLight(false);
+    setReverseLight(true);
+    setResetLight(true);
+  };
+
+  const reverseOrder = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+    setAscending(!ascending);
+    setSortAlphabeticallyLight(true);
+    setSortByLengthLight(true);
+    setReverseLight(false);
+    setResetLight(true);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetOrder = () => {
+    setGoods(initialGoods);
+    setAscending(true);
+    setSortAlphabeticallyLight(true);
+    setSortByLengthLight(true);
+    setReverseLight(true);
+    setResetLight(true);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortAlphabeticallyLight ? '' : 'is-light'}`}
+          onClick={sortAlphabetically}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${sortByLengthLight ? '' : 'is-light'}`}
+          onClick={sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${reverseLight ? '' : 'is-light'}`}
+          onClick={reverseOrder}
+        >
+          Reverse
+        </button>
+
+        <button
+          type="button"
+          className={`button is-danger ${resetLight ? '' : 'is-light'}`}
+          onClick={resetOrder}
+        >
+          Reset
+        </button>
+      </div>
+
+      <ul>
+        {goods.map((good, index) => (
```

Comment:
Using index as a key is not recommended. If the order of items changes, React will not be able to identify which items have changed, been added, or been removed. Consider using unique string as a key.